src/views/VideoView/VideoView.tsx

```diff
@@ -21,17 +22,25 @@ import { GetVideo, GetVideoVariables } from '@/api/queries/__generated__/GetVide
 import { formatVideoViewsAndDate } from '@/utils/video'
 import { AddVideoView, AddVideoViewVariables } from '@/api/queries/__generated__/AddVideoView'
 import { ChannelLink } from '@/components'
+import { useInterval, usePersonalData } from '@/hooks'
 
 const VideoView: React.FC<RouteComponentProps> = () => {
   const { id } = useParams()
   const { loading, data, error } = useQuery<GetVideo, GetVideoVariables>(GET_VIDEO, {
     variables: { id },
   })
   const [addVideoView] = useMutation<AddVideoView, AddVideoViewVariables>(ADD_VIDEO_VIEW)
+  const { state, updateWatchedVideos } = usePersonalData()
 
+  const videoTimestamp = useMemo(() => {
```

Comment:
I'm a bit concerned about that. _In theory_, React docs state that `useMemo` should be used only for performance and not as a semantic guarantee. _In theory_, future version of React could make `useMemo` recalculate the value from time to time. That would cause the video to start from the updated `startTime` again, right?

Speaking of which, now I'm wondering, `state.watchedVideos` is in the deps, won't that be changed during watching the video? When we do `updateWatchedVideos('INTERRUPTED', data.video.id, time)`?

Reply:
We can just run `useEffect once on mount here and save the timestamp to some state.